### DISCUSSION TITLE
Input-Leap stable-ish Protocol?

### DISCUSSION THREAD

**Discussion by whot** [shymega/input-leap-rs#14] [Q&A]
Just throwing this out as question: is there a plan to make the wire *protocol* a defined protocol? Because doing so would allow independent implementations of input-leap at the protocol barrier, e.g. tight integration of the protocol into gnome-shell or something like that.

**Comment by shymega** [comment]
Hey @whot. Sorry for the delay in my response.

This is something I haven't considered at this stage. However, the protocol is different to the C++ version. I intend for an efficient protocol for networking, and JSON-RPC for control of the 'primary' and the 'screens' on each computer.

Possibly might be using `msgpack` for inter-host communication.

The concepts are different from the C++ Input Leap, and the event system will be revamped too. There's a lot to decide at this stage. I was also thinking of integrating an 'RFC' type process for features, so we can have a formal verification and development process.

For now, there's a lot undecided.
